SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/Common/LinuxScripts.cs

```diff
@@ -43,14 +43,23 @@ exit 1
 WORKER_ID=$1
 shift 1
 
-# 2>/dev/null silences errors by redirecting stderr to the null device. This is done to prevent errors when a machine attempts to add the same user more than once.
+# Create a basic folder lock to ensure the user permissions modifications aren't run in parallel as otherwise they can race and fail
+if ! mkdir /improbable/logs/lockdir 2>/dev/null
```

Comment:
- Is this missing an rmdir after setup? 
- Could this be a loop instead waiting for mkdir to succeed?

Reply:
As `mkdir` is atomic, we can reliably know that only one script will be able to complete it successfully (it'll fail on all other calls, and that script will sleep to give the first script a change to complete the auth modifications). Not sure how you'd turn it into a loop, welcome suggestions.